src/go/rdctl/cmd/root.go

```diff
@@ -61,27 +62,39 @@ func Execute() {
 }
 
 func init() {
-  var err error
+	var err error
 
-  cobra.OnInitialize(initConfig)
-  configDir, err = os.UserConfigDir()
-  if err != nil {
-    log.Fatal("Can't get config-dir: ", err)
-  }
-  defaultConfigPath = filepath.Join(configDir, "rancher-desktop", "rd-engine.json")
-  rootCmd.PersistentFlags().StringVar(&configPath, "config-path", "", fmt.Sprintf("config file (default %s)", defaultConfigPath))
-  rootCmd.PersistentFlags().StringVar(&user, "user", "", "overrides the user setting in the config file")
-  rootCmd.PersistentFlags().StringVar(&host, "host", "", "default is localhost; most useful for WSL")
-  rootCmd.PersistentFlags().StringVar(&port, "port", "", "overrides the port setting in the config file")
-  rootCmd.PersistentFlags().StringVar(&password, "password", "", "overrides the password setting in the config file")
+	cobra.OnInitialize(initConfig)
+	configDir, err = os.UserConfigDir()
+	if err != nil {
+		log.Fatal("Can't get config-dir: ", err)
+	}
+	defaultConfigPath = filepath.Join(configDir, "rancher-desktop", "rd-engine.json")
+	rootCmd.PersistentFlags().StringVar(&configPath, "config-path", "", fmt.Sprintf("config file (default %s)", defaultConfigPath))
+	rootCmd.PersistentFlags().StringVar(&user, "user", "", "overrides the user setting in the config file")
+	rootCmd.PersistentFlags().StringVar(&host, "host", "", "default is localhost; most useful for WSL")
+	rootCmd.PersistentFlags().StringVar(&port, "port", "", "overrides the port setting in the config file")
+	rootCmd.PersistentFlags().StringVar(&password, "password", "", "overrides the password setting in the config file")
 }
 
-func doRequest(method string, command string)  error {
-  req, err := getRequestObject(method, command)
-  if err != nil {
-    return err
-  }
-  return doRestOfRequest(req)
+func doRequest(method string, command string) ([]byte, error) {
+	req, err := getRequestObject(method, command)
+	if err != nil {
+		return nil, err
+	}
+	return doRestOfRequest(req)
+}
+
+func doRequestWithPayload(method string, command string, payload *bytes.Buffer) ([]byte, error) {
+	req, err := http.NewRequest(method, fmt.Sprintf("http://%s:%s/%s/%s", host, port, apiVersion, command), payload)
```

Comment:
Should this just use `getRequestObject` (and make `doRequest` pass in `nil` for the body)?

If we need to override the content type, we can add `req.Header.Set("Content-Type", "application/json")`.

Reply:
Agreed. We can do that with the same refactoring moving the code into a `pkg` directory.